src/java.base/windows/classes/sun/nio/ch/FileKey.java

```diff
@@ -47,9 +47,7 @@ public static FileKey create(FileDescriptor fd) throws IOException {
 
     @Override
     public int hashCode() {
-        return (int)(dwVolumeSerialNumber ^ (dwVolumeSerialNumber >>> 32)) +
-               (int)(nFileIndexHigh ^ (nFileIndexHigh >>> 32)) +
-               (int)(nFileIndexLow ^ (nFileIndexLow >>> 32));
+        return dwVolumeSerialNumber + nFileIndexHigh + nFileIndexLow;
```

Comment:
I assume that that it's not too common to use FileChannel.lock to lock many regions of a file so the quality of the hashCode might not b important. However, looking at it now makes me wonder about the addition, maybe it should do the usual multiple by prime and add the next common, as if calling Objects.hashCode.

Reply:
So changed in 414fad626d812a6b3b8976816f6351125f7481bf.